pom.xml

```diff
@@ -289,6 +289,28 @@ under the License.
         </configuration>
       </plugin>
 
+      <plugin>
```

Comment:
According to [Maven Checkstyle Plugin documentation](https://maven.apache.org/plugins/maven-checkstyle-plugin/examples/upgrading-checkstyle.html), one can use newer version of Checkstyle by specifying `com.puppycrawl.tools` dependency version. Right now, Checkstyle has version 10.3.1 (instead of "default" 8.29 used by Maven Checkstyle Plugin).

Reply:
yes but 10.3.1 is not available in InteliJ checkstyle plugin configuration and checkstyle.xml that we are using may not be compatible with 10.3.1.

Reply:
Ok, fair point :) Just wanted to point it out, as it shows such an information when doing `mvn verify checkstyle:checkstyle`. But it's no big deal, it can be as it is right now.